wip/Manage Tasks and Variants

ksmontville opened the pull request:
This PR builds out the initial components for managing tasks and variants through the Dashboard. Requires an updated version of Firekit which enables these methods.

ksmontville commented:
Uses the gameConfig object stored in the task document to facilitate the creation of new variants

![image](https://github.com/yeatmanlab/roar-dashboard/assets/100953047/9e055703-4e00-4b07-b6da-7e306156db8a)

ksmontville commented:
![image](https://github.com/yeatmanlab/roar-dashboard/assets/100953047/8eb7975d-731a-44ab-b077-084acd79bfbd)
![image](https://github.com/yeatmanlab/roar-dashboard/assets/100953047/2f77019b-0b2a-460b-82b3-1ea857b4237f)
![image](https://github.com/yeatmanlab/roar-dashboard/assets/100953047/bcecd29d-69bb-445d-b121-502193379509)

Emily-ejag reviewed (changes requested):
The code looks good, but maybe adding a handler in the dropdown on selected variants, because it is showing empty variants: 
<img width="292" alt="image" src="https://github.com/yeatmanlab/roar-dashboard/assets/62304493/36cdab11-6206-4e14-b6e1-f6ec23c49309">

Emily-ejag commented:
Also If I add a new Param and called the same thing as an existing, it overwrites the value, for example I changed "consent" by adding a new param, and switch from boolean to string, I suggest a error handling here is the params have the same name, to not overwrite: 
<img width="497" alt="image" src="https://github.com/yeatmanlab/roar-dashboard/assets/62304493/dc1ca088-4a70-4ec8-8b42-917527c2b338">

lucasxsong commented on src/components/tasks/ManageTasks.vue:
```diff
@@ -0,0 +1,617 @@
+          </div>
+
+          <div v-if="!isExternalTask">
+            <h3 class="text-center">Configure Game Parameters</h3>
```
Could I suggest that this be bolded to match the other header?

lucasxsong commented on src/components/tasks/ManageVariants.vue:
```diff
@@ -0,0 +1,687 @@
+      </div>
+    </PvTabPanel>
+
+    <PvTabPanel header="Update Variant">
```
I feel that we could probably find a better method to display the switch between `update/create` for variants and tasks than nested panels. I know that you had mentioned that you were planning on switching. Could I suggest the use of [radio buttons ](https://primevue.org/radiobutton/) to toggle between the two, similar to how @Emily-ejag uses them to toggle between the different modes of consent picking?

ksmontville replied:
I agree that it isn't ideal, but I think that I will hold off on any additional changes until we receive some feedback from users. For now, I made sure that the nested tabs are entirely navigable by keyboard actions.

lucasxsong reviewed (changes requested):
This is looking good and will be a great feature add to the dashboard! I had a few comments, let me know if you'd like to talk about anything.

ksmontville commented on src/components/tasks/ManageTasks.vue:
```diff
@@ -0,0 +1,617 @@
+          <div class="flex flex-column row-gap-3">
+            <section class="form-section">
+              <div class="p-input-icon-right">
+                <label for="taskName">Task Name <span class="required">*</span></label>
```
Good idea

ksmontville commented on src/components/tasks/ManageVariants.vue:
```diff
@@ -0,0 +1,687 @@
+    taskImage: variantFields.selectedGame.image,
+    variantName: variantFields.variantName,
+    variantParams: combinedParams,
+    // TODO: Check if this is the valid way to see demo/test data values
```
Yes, removed!

ksmontville commented on src/components/tasks/ManageVariants.vue:
```diff
@@ -0,0 +1,687 @@
+        <section v-if="selectedVariant" class="flex flex-column align-items-start mt-4">
+          <div class="flex flex-column w-full">
+            <label for="fieldsOutput">
+              <strong>Fields</strong>
```
I've set `w-2` and `text-center` on the disabled type input for now, let me know what you think.

ksmontville commented:
> Also If I add a new Param and called the same thing as an existing, it overwrites the value, for example I changed "consent" by adding a new param, and switch from boolean to string, I suggest a error handling here if the params have the same name, to not overwrite: <img alt="image" width="497" src="https://private-user-images.githubusercontent.com/62304493/336167943-dc1ca088-4a70-4ec8-8b42-917527c2b338.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc3Nzg5OTMsIm5iZiI6MTcxNzc3ODY5MywicGF0aCI6Ii82MjMwNDQ5My8zMzYxNjc5NDMtZGMxY2EwODgtNGE3MC00ZWM4LThiNDItOTE3NTI3YzJiMzM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA3VDE2NDQ1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYyYWY5OTI1Nzk3MWQwYmNhYWExM2Q5Y2Q3NDZmZjUwYWI0NTFhMjZlM2NmZTI0MWRhNmNlYTUyYzQ1YmM5ZDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.HAKcHa6pUrn1yZwIOvlfEjkgybj0VEKGxc-Bue1hnsA">

This is a great catch, thank you.

I have added some validation that should address this.
